app/uk/gov/hmrc/tai/service/TaxAccountSummaryService.scala

```diff
@@ -42,36 +42,42 @@ class TaxAccountSummaryService @Inject()(
   def taxAccountSummaryViewModel(nino: Nino, taxAccountSummary: TaxAccountSummary)(
     implicit hc: HeaderCarrier,
     messages: Messages): Future[TaxAccountSummaryViewModel] =
-    for {
-      livePensionIncomeSources      <- taxAccountService.incomeSources(nino, TaxYear(), PensionIncome, Live)
-      liveEmploymentIncomeSources   <- taxAccountService.incomeSources(nino, TaxYear(), EmploymentIncome, Live)
-      ceasedEmploymentIncomeSources <- taxAccountService.incomeSources(nino, TaxYear(), EmploymentIncome, NotLive)
-      nonMatchingCeasedEmployments  <- employmentService.ceasedEmployments(nino, TaxYear())
-      nonTaxCodeIncome              <- taxAccountService.nonTaxCodeIncomes(nino, TaxYear())
-      isAnyFormInProgress           <- trackingService.isAnyIFormInProgress(nino.nino)
-    } yield {
-      (
-        livePensionIncomeSources,
-        liveEmploymentIncomeSources,
-        ceasedEmploymentIncomeSources,
-        nonMatchingCeasedEmployments,
-        nonTaxCodeIncome) match {
-        case (
-            TaiSuccessResponseWithPayload(livePensionIncomeSources: Seq[TaxedIncome]),
-            TaiSuccessResponseWithPayload(liveEmploymentIncomeSources: Seq[TaxedIncome]),
-            TaiSuccessResponseWithPayload(ceasedEmploymentIncomeSources: Seq[TaxedIncome]),
-            nonMatchingCeasedEmployments: Seq[Employment],
-            TaiSuccessResponseWithPayload(nonTaxCodeIncome: NonTaxCodeIncome)) =>
-          TaxAccountSummaryViewModel(
-            taxAccountSummary,
-            isAnyFormInProgress,
-            nonTaxCodeIncome,
-            IncomeSources(livePensionIncomeSources, liveEmploymentIncomeSources, ceasedEmploymentIncomeSources),
-            nonMatchingCeasedEmployments
-          )
-        case (_, _, _, _, TaiUnauthorisedResponse(message)) => throw new UnauthorizedException(message)
-        case _                                              => throw new RuntimeException("Failed to fetch income details")
-      }
+    (
+      taxAccountService.incomeSources(nino, TaxYear(), PensionIncome, Live),
+      taxAccountService.incomeSources(nino, TaxYear(), EmploymentIncome, Live),
+      taxAccountService.incomeSources(nino, TaxYear(), EmploymentIncome, NotLive),
+      employmentService.ceasedEmployments(nino, TaxYear()),
+      taxAccountService.nonTaxCodeIncomes(nino, TaxYear()),
+      trackingService.isAnyIFormInProgress(nino.nino)).mapN {
+      case (
+          livePensionIncomeSources,
+          liveEmploymentIncomeSources,
+          ceasedEmploymentIncomeSources,
+          nonMatchingCeasedEmployments,
+          nonTaxCodeIncome,
+          isAnyFormInProgress) =>
+        (
+          livePensionIncomeSources,
+          liveEmploymentIncomeSources,
+          ceasedEmploymentIncomeSources,
+          nonMatchingCeasedEmployments,
+          nonTaxCodeIncome) match {
```

Comment:
I think this can be reduced to
```def taxAccountSummaryViewModel(nino: Nino, taxAccountSummary: TaxAccountSummary)(
    implicit hc: HeaderCarrier,
    messages: Messages): Future[TaxAccountSummaryViewModel] =
    (
      taxAccountService.incomeSources(nino, TaxYear(), PensionIncome, Live),
      taxAccountService.incomeSources(nino, TaxYear(), EmploymentIncome, Live),
      taxAccountService.incomeSources(nino, TaxYear(), EmploymentIncome, NotLive),
      employmentService.ceasedEmployments(nino, TaxYear()),
      taxAccountService.nonTaxCodeIncomes(nino, TaxYear()),
      trackingService.isAnyIFormInProgress(nino.nino)).mapN {
          case (
              TaiSuccessResponseWithPayload(livePensionIncomeSources: Seq[TaxedIncome]),
              TaiSuccessResponseWithPayload(liveEmploymentIncomeSources: Seq[TaxedIncome]),
              TaiSuccessResponseWithPayload(ceasedEmploymentIncomeSources: Seq[TaxedIncome]),
              nonMatchingCeasedEmployments: Seq[Employment],
              TaiSuccessResponseWithPayload(nonTaxCodeIncome: NonTaxCodeIncome),
              isAnyFormInProgress) =>
            TaxAccountSummaryViewModel(
              taxAccountSummary,
              isAnyFormInProgress,
              nonTaxCodeIncome,
              IncomeSources(livePensionIncomeSources, liveEmploymentIncomeSources, ceasedEmploymentIncomeSources),
              nonMatchingCeasedEmployments
            )
          case (_, _, _, _, TaiUnauthorisedResponse(message), _) => throw new UnauthorizedException(message)
          case _                                              => throw new RuntimeException("Failed to fetch income details")
        }
    }```